src/components/PromiseList/PromiseCard.svelte

```diff
@@ -1,7 +1,17 @@
 <script lang="ts">
 	import { formatThaiDate } from '$lib/date-parser';
-	import { PromiseStatus, type PromiseSummary } from '$models/promise';
 	import Quotes from 'carbon-icons-svelte/lib/Quotes.svelte';
+	import { PromiseStatus } from '$models/promise';
+
+	type PromiseSummary = {
+		id: string;
+		status: PromiseStatus;
+		latestProgressDate?: Date;
+		party: { name: string; logo: string };
+		statements: string[];
+		keywords: string[];
+		categories: string[];
+	};
 
 	export let promiseSummary: PromiseSummary;
```

Comment:
I think it is unnessesary to nest those property into another object as promiseSummary prop. It make code more complicated from both inside this component and any other components that might use this one. (It used to makesense that way because we tied it to the model, but now we are getting rid of it).

I suggest you to extract id, status, etc. to be seperated props then remove the promiseSummary.